pkgs/development/libraries/glib/setup-hook.sh

```diff
@@ -1,3 +1,10 @@
+glibAddGioExtraModules() {
+    if [ -d "$1/lib/gio/modules" ] && [ -n "$(ls -A "$1/lib/gio/modules")" ] ; then
+        addToSearchPath GIO_EXTRA_MODULES "$1/lib/gio/modules"
+    fi
+}
+addEnvHooks "$hostOffset" glibAddGioExtraModules
```

Comment:
IIUC, the purpose of this is to make it work in nix-shell?

I think we might want to unify this with `wrapGAppsHook` so that we do not need to do the [wrapper hack](https://discourse.nixos.org/t/python-qt-woes/11808/2) for other variables.

Reply:
with this change i am trying to prevent regressions where the binaries in  gst-plugins-base and gstreamer using https streams. since the libgstsoup.so plugin will work it requires the glib-networking module for https.

today, if you do:
```console
nix-shell -p gst_all_1.gst-plugins-good gst_all_1.gst-plugins-base gst_all_1.gstreamer gst_all_1.gst-plugins-ugly
gst-play-1.0 https://path/to/some.mp3
```
that will work as it uses the Curl plugin from the ugly.

but once i fix the loading of libsoup_3 it will fail as the libgstsoup.so plugin will be loaded from good but will fail to find glib-networking unless the GST_EXTRA_MODULES is set properly.

Reply:
I misread your question the first time around...
> IIUC, the purpose of this is to make it work in nix-shell?

yes.

> I think we might want to unify this with `wrapGAppsHook` so that we do not need to do the [wrapper hack](https://discourse.nixos.org/t/python-qt-woes/11808/2) for other variables.

I don't understand this. today `nix-shell` and `nix develop` will have the required environment variables GST_PLUGIN_SYSTEM_PATH_1_0 so things like gst-play-1.0 will work. however, `nix shell` will not but `gst-play-`1.0` is already broken as it will fail to find any plugins.

i'm experimenting with adding a buildEnv to gstreamer which could wrap `gst-play-1.0` so `nix shell` could work eg:
```
nix shell --impure --expr 'with import ./. {}; gst_all_1.buildEnv { paths = with gst_all_1; [ gst-plugins-good gst-plugins-base gst-libav gstreamer ]; }'
```

maybe that's what you mean by using `wrapGAppsHook`?

Reply:
I meant that `wrapGAppsHook` does not set most of the environment variables in the builder, only pass them to the wrapper, so they are not available to the shell.

Reply:
> I meant that `wrapGAppsHook` does not set most of the environment variables in the builder, only pass them to the wrapper, so they are not available to the shell.

i'm still missing the core concept here. if the environment variables do not exist, then the gstreamer binaries will not work in nix-shell.

Reply:
Right, I was talking about the current state, where most variables are not available in the shell but [some are](https://github.com/NixOS/nixpkgs/blob/d2a5f168590d7c5b3c4aa219a12a269ad7d703a9/pkgs/build-support/setup-hooks/wrap-gapps-hook/wrap-gapps-hook.sh#L30-L34). I was bringing it up since it might be worth considering doing the same for the rest of the variables.

Reply:
this is still ambiguous. what are you trying to accomplish -- moving everything to env vars or moving away from using env vars?

Reply:
anyway, i did not modify [wrap-gapps-hook.sh](https://github.com/NixOS/nixpkgs/blob/d2a5f168590d7c5b3c4aa219a12a269ad7d703a9/pkgs/build-support/setup-hooks/wrap-gapps-hook/wrap-gapps-hook.sh) to use the `GIO_EXTRA_MODULES` out of the environment when setting up the wrapper args just because I am not sure if exporting `GIO_EXTRA_MODULES` is a good thing to do in the builder.

exporting GIO_EXTRA_MODULES in the builder means that any program that uses glib in the builder after the env var is exported will load the modules, which seems undesirable.

if that seems like a non-issue then yes, i will update wrap-gapps-hook.sh to remove the duplicated code.